externals/kyuubi-trino-engine/pom.xml

```diff
@@ -85,4 +84,72 @@
 
     </dependencies>
 
+    <build>
+        <outputDirectory>target/scala-${scala.binary.version}/classes</outputDirectory>
+        <testOutputDirectory>target/scala-${scala.binary.version}/test-classes</testOutputDirectory>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
```

Comment:
Some thoughts on shade engine.

Shade always should be the latest approach to solve the problems. Shade and class relocation makes it difficult to debug, and when `NoClassFoundException` or `NoSuchMethodException` occurs, it's not easy to dig which it comes from which jars. Class relocation also may breaks the reflection. There are some bugs on `maven-shade-plugin` which may hang the compile.

We shade Spark engine and Flink engine because we need to submit the jar to run with the engine runtime jars, so there is a risk which may cause class conflict, because we don't know which jars user put into their engine runtime jar folder.

For Trino engine, I think we don't need to run it with Trino runtime jars. A simple layout e.g. `bin/`, `conf/`, `lib/` maybe a better choice.

Reply:
Agree